src/mario_pouch.c

```diff
@@ -306,167 +306,477 @@ asm void pouchRemoveShopItemIdx(s16 itemId, s32 idx)
     #include "asm/8014edb8.s"
 }
 
-asm PouchCharOrPixlInfo * pouchGetCharInfo(s32 slot)
+PouchCharOrPixlInfo * pouchGetCharInfo(s32 slot)
 {
-    #include "asm/8014ef98.s"
+    MarioPouchWork * pp;
+
+    pp = pouchGetPtr();
+
+    return &pp->characters[slot];
 }
 
-asm void pouchMakeCharSelectable(s16 itemId)
+void pouchMakeCharSelectable(s16 id)
 {
-    #include "asm/8014efb0.s"
+    MarioPouchWork * pp;
+    s32 i;
+
+    pp = pouchGetPtr();
+
+    for (i = 0; i < 4; i++)
+        if (id == pp->characters[i].id)
+            break;
+
+    if (i < 4)
+        pp->characters[i].selectable = true;
 }
 
-asm void pouchMakeCharNotSelectable(s16 itemId)
+void pouchMakeCharNotSelectable(s16 id)
 {
-    #include "asm/8014f018.s"
+    MarioPouchWork * pp;
+    s32 i;
+
+    pp = pouchGetPtr();
+
+    for (i = 0; i < 4; i++)
+        if (id == pp->characters[i].id)
+            break;
+
+    if (i < 4)
+        pp->characters[i].selectable = false;
 }
 
-asm PouchCharOrPixlInfo * pouchGetPixlInfo(s32 slot)
+PouchCharOrPixlInfo * pouchGetPixlInfo(s32 slot)
 {
-    #include "asm/8014f080.s"
+    MarioPouchWork * pp;
+    
+    pp = pouchGetPtr();
+
+    return &pp->pixls[slot];
 }
 
-asm void pouchMakePixlSelectable(s16 itemId)
+void pouchMakePixlSelectable(s16 id)
 {
-    #include "asm/8014f098.s"
+    MarioPouchWork * pp;
+    s32 i;
+
+    pp = pouchGetPtr();
+
+    for (i = 0; i < 16; i++)
+        if (id == pp->pixls[i].id)
+            break;
+    
+    if (i < 16)
+        pp->pixls[i].selectable = true;
 }
 
-asm void pouchMakePixlNotSelectable(s16 itemId)
+void pouchMakePixlNotSelectable(s16 id)
 {
-    #include "asm/8014f154.s"
+    MarioPouchWork * pp;
+    s32 i;
+
+    pp = pouchGetPtr();
+
+    for (i = 0; i < 16; i++)
+        if (id == pp->pixls[i].id)
+            break;
+    
+    if (i < 16)
+        pp->pixls[i].selectable = false;
 }
 
-asm bool pouchCheckPixlSelected(s16 id)
+bool pouchCheckPixlSelected(s16 id)
 {
-    #include "asm/8014f210.s"
+    MarioPouchWork * pp;
+    s32 i;
+
+    pp = pouchGetPtr();
+
+    for (i = 0; i < 16; i++)
+    {
+        PouchCharOrPixlInfo * p = &pp->pixls[i];
+        if (p->id != NULL && p->id == id)
```

Comment:
NULL doesn't really make sense here since it's not a pointer, could be a good time to start the actual item id enum and use ITEM_ID_NULL

Reply:
I'm used to C++'s `nullptr` for that. I'll work on the enum tonight.